sonar-auth-saml-plugin/src/main/java/org/sonarsource/auth/saml/SamlIdentityProvider.java

```diff
@@ -66,11 +95,91 @@ public boolean allowsUsersToSignUp() {
 
   @Override
   public void init(InitContext context) {
-    // TODO
+    try {
+      Auth auth = newAuth(initSettings(context.getCallbackUrl()), context.getRequest(), context.getResponse());
+      auth.login();
+    } catch (IOException | SettingsException e) {
+      throw new IllegalStateException("Fail to init", e);
```

Comment:
I would have put a more descriptive error : "Fail to intialize SAML authentication plugin" for instance.